python/labbox_ephys/import_to_workspace.py

```diff
@@ -0,0 +1,134 @@
+import spikeextractors as se
+import numpy as np
+import labbox_ephys as le
+import kachery_p2p as kp
+import kachery as ka
+
+def _get_recordings_from_subfeed(recordings_subfeed: kp.Subfeed):
+    le_recordings = {}
+    while True:
+        msg = recordings_subfeed.get_next_message(wait_msec=0)
+        if msg is None: break
+        if 'action' in msg:
+            a = msg['action']
+            if a.get('type', '') == 'ADD_RECORDING':
+                r = a.get('recording', {})
+                rid = r.get('recordingId', '')
+                le_recordings[rid] = r
+            elif a.get('type', '') == 'DELETE_RECORDINGS':
+                for rid in a.get('recordingIds', []):
+                    if rid in le_recordings:
+                        del le_recordings[rid]
+    return le_recordings
+
+def _get_sortings_from_subfeed(sortings_subfeed: kp.Subfeed):
+    le_sortings = {}
+    while True:
+        msg = sortings_subfeed.get_next_message(wait_msec=0)
+        if msg is None: break
+        if 'action' in msg:
+            a = msg['action']
+            if a.get('type', '') == 'ADD_SORTING':
+                s = a.get('sorting', {})
+                sid = s.get('sortingId', '')
+                le_sortings[sid] = s
+            elif a.get('type', '') == 'DELETE_SORTINGS':
+                for sid in a.get('sortingIds', []):
+                    if sid in le_sortings:
+                        del le_sortings[sid]
+    return le_sortings
+
+def _import_le_recording(recordings_subfeed: kp.Subfeed, le_recording):
+    recordings_subfeed.set_position(0)
+    le_recordings = _get_recordings_from_subfeed(recordings_subfeed)
+    id = le_recording['recordingId']
+    if id in le_recordings:
+        print(f'Recording with ID {id} already exists. Not adding.')
+        return
+    print(f'Adding recording: {id}')
+    recordings_subfeed.submit_message({
+        'action': {
+            'type': 'ADD_RECORDING',
+            'recording': le_recording
+        }
+    })
+
+def _import_le_sorting(sortings_subfeed: kp.Subfeed, le_sorting):
+    sortings_subfeed.set_position(0)
+    le_sortings = _get_sortings_from_subfeed(sortings_subfeed)
+    id = le_sorting["sortingId"]
+    if id in le_sortings:
+        print(f'Sorting with ID {id} already exists. Not adding.')
+        return
+    print(f'Adding sorting: {id}')
+    sortings_subfeed.submit_message({
+        'action': {
+            'type': 'ADD_SORTING',
+            'sorting': le_sorting
+        }
+    })
+
+def import_recording(*, feed: kp.Feed, workspace_name: str, recording: se.RecordingExtractor, recording_id: str):
```

Comment:
Import a recording into a workspace. This is to be accessed in a Python script - outside of the GUI. Snippets in the GUI will refer to this function.